api/product.rb

```diff
@@ -101,7 +104,17 @@ def version_obj(name)
     end
 
     def version_obj_or_default(name)
```

Comment:
I don't think this method has anything to do with the `default` version any more. In which case it should be renamed.

Reply:
Agreed. It seems like `version_obj` and `version_obj_or_default` should be consolidated into something like `version`. Call it and it returns the proper version object for you or errors.